snapshots/overlay/overlay_test.go

```diff
@@ -335,12 +336,30 @@ func testOverlayView(t *testing.T, newSnapshotter testsuite.SnapshotterFunc) {
 	if m.Source != "overlay" {
 		t.Errorf("mount source should be overlay but received %q", m.Source)
 	}
-	if len(m.Options) != 2 {
-		t.Errorf("expected 1 additional mount option but got %d", len(m.Options))
+	if getIndexOff() {
+		if len(m.Options) != 2 {
+			t.Errorf("expected 1 additional mount option but got %d", len(m.Options))
+		}
+		lowers := getParents(ctx, o, root, "/tmp/view2")
+		expected = fmt.Sprintf("lowerdir=%s:%s", lowers[0], lowers[1])
+		if m.Options[1] != expected {
+			t.Errorf("expected option %q but received %q", expected, m.Options[0])
+		}
+	} else {
+		if len(m.Options) != 1 {
+			t.Errorf("expected 1 additional mount option but got %d", len(m.Options))
+		}
+		lowers := getParents(ctx, o, root, "/tmp/view2")
+		expected = fmt.Sprintf("lowerdir=%s:%s", lowers[0], lowers[1])
+		if m.Options[0] != expected {
+			t.Errorf("expected option %q but received %q", expected, m.Options[0])
+		}
 	}
-	lowers := getParents(ctx, o, root, "/tmp/view2")
-	expected = fmt.Sprintf("lowerdir=%s:%s", lowers[0], lowers[1])
-	if m.Options[1] != expected {
-		t.Errorf("expected option %q but received %q", expected, m.Options[0])
+}
+func getIndexOff() bool {
```

Comment:
Unless this can change often at runtime, I would like to see this bool be wrapped in a `sync.Once` so that we don't hit sysfs many times.  I think there are other areas in the snapshotters that use `sync.Once` as an example.

Reply:
var indexOff bool
var indexOffOnce sync.Once
func getIndexOff() {
	if _, err := os.Stat("/sys/module/overlay/parameters/index"); err == nil {
		indexOff = true
	}
}
like this ?

Reply:
can these code ok ?
var indexOff bool
var indexOffOnce sync.Once
func getIndexOff() {
	if _, err := os.Stat("/sys/module/overlay/parameters/index"); err == nil {
		indexOff = true
	}
}